kv/store.go

```diff
@@ -49,11 +50,25 @@ type Value interface {
 	Version() int
 }
 
+// ValueWatch provides updates to a Value
+type ValueWatch interface {
+	// C returns the notification channel
+	C() <-chan struct{}
+	// Get returns the latest version of the value
+	Get() Value
+	// Close stops watching for value updates
+	Close()
+}
+
 // Store provides access to the configuration store
 type Store interface {
 	// Get retrieves the value for the given key
 	Get(key string) (Value, error)
 
+	// Watch adds a watch for value updates for given key. Initial value
+	// will be available via ValueWatch.Get()
+	Watch(key string) (ValueWatch, error)
```

Comment:
We should call out explicitly whether this will block until the initial value is available.